torch_tvm/custom_tvm_ops/topi/topi_init.cc

```diff
@@ -0,0 +1,62 @@
+#include <topi/x86/default.h>
+#include <tvm/build_module.h>
+#include <tvm/relay/attrs/transform.h>
+#include <tvm/relay/op_attr_types.h>
+#include <tvm/relay/expr.h>
+#include <tvm/runtime/packed_func.h>
+
+#include <custom_tvm_ops/relay/layer_norm_attrs.h>
+
+#include "topi_init.h"
+#include "topi/generic/injective.h"
+#include "custom_layer_norm_generic_sched.h"
+#include "custom_layer_norm.h"
+
+#include <mutex>
+
+namespace tvm {
+
+using tvm::relay::OpPatternKind;
+
+void registerCustomTopiSchedules() {
```

Comment:
can you use static init instead of std::once?

Reply:
@bwasti, To be able encapsulate all this in a function is kind of nice. And if it is wrapped in a function I would think call_once is a good way to ensure single initialization. I am open to suggestions.

Reply:
Sorry, I wasn't clear - you can use a static initialization approach rather than calling the function repeatedly and guarding the execution with std::once.  The idea boils down to defining a static variable whose constructor does the registration.

You can find this paradigm littered throughout PyTorch, and might even be able to use some of the helpers https://bddppq.github.io/codebrowser/pytorch/pytorch/c10/util/Registry.h.html#5
https://bddppq.github.io/codebrowser/pytorch/pytorch/torch/csrc/jit/custom_operator.h.html#10
https://bddppq.github.io/codebrowser/pytorch/pytorch/torch/csrc/jit/pass_manager.h.html#9

Reply:
So I have seen these in PyTorch codebase, and I think `auto reg_ptr = tvm::runtime::Registry::Get("relay.op._Register");` does something similar. So in effect this is doing the same thing, except it ensure we dont call this method more than once. 
If you want to remove call_once, it would just mean that we remove all the function boundaries so that code is executed when the lib is loaded. We can discuss this further if you think there is a more elegant approach, but I am not convinced of it or see it yet.

Reply:
let's discuss offline

Reply:
I agree with @bwasti with the static trick.